src/cpp/lpnamer/model/Problem.h

```diff
@@ -0,0 +1,108 @@
+//
+// Created by marechaljas on 27/04/2022.
+//
+
+#ifndef ANTARESXPANSION_SRC_CPP_LPNAMER_MODEL_PROBLEM_H_
+#define ANTARESXPANSION_SRC_CPP_LPNAMER_MODEL_PROBLEM_H_
+
+#include <utility>
+
+#include "multisolver_interface/SolverAbstract.h"
+
+/**
+ * @brief Decorator to the SolverAbstract class
+ */
+class Problem: public SolverAbstract
+{
+ public:
+  Problem() = delete;
+  explicit Problem(SolverAbstract::Ptr solver_abstract)
+  : solver_abstract_(std::move(solver_abstract))
+  {
+  }
+  private:
+  const SolverAbstract::Ptr solver_abstract_;
```

Comment:
Problem is a decorator.
It decorates a object of the SolverAbstract hierarchy with the mc_year property (and read_prob_mps). That means an instance of Problem can decorate any of SolverCBC, SolverCLP or SolverXpress without altering the behaviour of those class.
It can't be pure virtual since it need to be instantiated.
It does override the methods but more than this, it forwards the behaviour to the SolverAbstract instance it decorates.